### PR TITLE
[RFC] MSVC: Define S_IXUSR.

### PR DESCRIPTION
**Pull request by sethjackson**
Cherry picked from #810. From equalsraf@34eea58.

### CODE REVIEW

File: src/nvim/os/win_defs.h [hunk 1]
```diff
@@ -15,6 +16,15 @@
...
 # endif
+# ifndef S_ISDIR
+#  define S_ISDIR(mode) (((mode) & S_IFMT) == S_IFDIR)
+# endif
+# ifndef S_ISREG
+#  define S_ISREG(mode) (((mode) & S_IFMT) == S_IFREG)
+# endif
+# ifndef S_IXUSR
+#  define S_IXUSR S_IEXEC
+# endif
```
Comment (justinmk) [inline src/nvim/os/win_defs.h]: Unless I am mistaken the only one that is needed is this one. The others are here:

https://github.com/neovim/neovim/blob/master/src/nvim/os/os_defs.h#L54-L71

Comment (sethjackson) [inline src/nvim/os/win_defs.h]: Oh right. Thanks! Updated. :)

**Comment by justinmk** [comment]
Thanks @sethjackson. 

@equalsraf When you have a moment, could you remove the `S_ISDIR/S_ISREG/S_IXUSR` defines from  https://github.com/equalsraf/neovim/commit/34eea58 ? They don't conflict during rebase because they were added to os_defs in another PR.

**Comment by equalsraf** [comment]
@justinmk done with the latest rebase.

**Pull request merged**